internal/internal_utils.go

```diff
@@ -203,11 +204,20 @@ func getErrorDetails(err error, dataConverter encoded.DataConverter) (string, []
 		}
 		return errReasonPanic, data
 	case *TimeoutError:
-		data, err0 := encodeArgs(dataConverter, []interface{}{err.timeoutType})
+		var data []byte
+		var err0 error
+		switch details := err.details.(type) {
+		case ErrorDetailsValues:
+			data, err0 = encodeArgs(dataConverter, details)
+		case *EncodedValues:
+			data = details.values
+		default:
+			panic("unknown error type")
+		}
 		if err0 != nil {
 			panic(err0)
 		}
-		return errReasonTimeout, data
+		return fmt.Sprintf("%v %v", errReasonTimeout, err.timeoutType), data
```

Comment:
Use the same format as server side: https://github.com/uber/cadence/blob/master/service/history/retry.go#L117